src/main.rs

```diff
@@ -92,22 +92,29 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     connect(&mut rich_presence_client);
     println!("{}\n{}", "Connected to Discord Rich Presence Socket".bright_green().bold(), "------------------------------------------------------------------".bold());
 
+    // Keeps track of the previous content item to compare later.
+    let mut last_content_id: String = "".to_string();
+
     // Start loop
     loop {
         let mut content = get_jellyfin_playing(&config.url, &config.api_key, &config.username, &config.enable_images).await?;
 
         if !content.media_type.is_empty() {
             // Print what we're watching
             if !connected {
-                println!("{}\n{}", content.details.bright_cyan().bold(), content.state_message.bright_cyan().bold());
-
                 // Set connected to true so that we don't try to connect again
                 connected = true;
             }
             if config.imgur_images {
                 content.image_url = get_image_imgur(&content.image_url, &content.item_id, &config.imgur_client_id, args.image_urls.clone()).await?;
             }
 
+            // Compare the current content to the previous one to see if it's different. Dump it to console if it is.
+            if content.individual_item_id != last_content_id{
+                last_content_id = content.individual_item_id;
+                println!("\n{}\n{}", content.details.bright_cyan().bold(), content.state_message.bright_cyan().bold());
+            }
+            
```

Comment:
Do we have to check this to avoid printing the same information again? It's a lot of logic to run every single loop

Reply:
Nah, that can be removed if you don't think it's necessary. However, the only actual logic happening is one string comparison as the ID is returned from the call to get current watching.

Reply:
I was thinking more of how its querying jellyfin for information it only might need. I think its better to just let it print the info again once reconnected to avoid making this more complicated than it already is. Apart from that your fix looks great.

Reply:
It hasn't actually queried Jellyfin for any extra info though, as it's already in the response object. I don't think adding that to the struct and returning an extra value would add too much overhead would it?

I realize the feature isn't exactly required though.

Reply:
No, it wouldn't. I would just like to keep it simple, its true that it is already in the response object. I was just thinking about low powered devices, ram usage and cpu usage. This program has a very simple task and I already think its getting quite bloated because I'm not very experienced with rust yet. We can keep it in there if you want but I think it adds unneeded complexity just to avoid an extra print.

Reply:
Also by the way i hope you know you arent crediting yourself when you make commits, your git username and email arent set to your github ones

Reply:
Removed the console changes now.

Also, thanks didn't realize. This is the first public repository I've contributed to so didn't notice.